subsys/bluetooth/host/classic/a2dp.c

```diff
@@ -76,35 +118,688 @@ struct bt_a2dp *get_new_connection(struct bt_conn *conn)
 	}
 
 	/* Clean the memory area before returning */
-	a2d_reset(&connection[free]);
+	a2dp_reset(&connection[free]);
 
 	return &connection[free];
 }
 
-int a2dp_accept(struct bt_conn *conn, struct bt_avdtp **session)
+void avdtp_connected(struct bt_avdtp *session)
 {
-	struct bt_a2dp *a2dp_conn;
+	struct bt_a2dp *a2dp = A2DP_AVDTP(session);
 
-	a2dp_conn = get_new_connection(conn);
-	if (!a2dp_conn) {
-		return -ENOMEM;
+	a2dp->a2dp_state = INTERNAL_STATE_AVDTP_CONNECTED;
+	/* notify a2dp app the connection. */
+	if ((a2dp_cb != NULL) && (a2dp_cb->connected != NULL)) {
+		a2dp_cb->connected(a2dp, 0);
+	}
+}
+
+void avdtp_disconnected(struct bt_avdtp *session)
+{
+	struct bt_a2dp *a2dp = A2DP_AVDTP(session);
+
+	/* notify a2dp app the disconnection. */
+	if ((a2dp_cb != NULL) && (a2dp_cb->disconnected != NULL)) {
+		a2dp_cb->disconnected(a2dp);
+	}
+}
+
+static struct net_buf *avdtp_alloc_buf(struct bt_avdtp *session)
+{
+	return NULL;
+}
+
+int a2dp_discovery_ind(struct bt_avdtp *session, uint8_t *errcode)
+{
```

Comment:
This function seems to do nothing？？

Reply:
It is AVDTP interface callback, A2DP return success/accept directly to let avdtp layer to response the AVDTP_DISCOVER cmd directly. I think A2DP don't need to call back this event to app because the codec endpoints are registered from application, the AVDTP_DISCOVER cmd can be responded directly.